pkg/controller/backup_repository_controller.go

```diff
@@ -130,18 +127,41 @@ func (r *BackupRepoReconciler) invalidateBackupReposForBSL(ctx context.Context,
 		return []reconcile.Request{}
 	}
 
+	requests := []reconcile.Request{}
 	for i := range list.Items {
 		r.logger.WithField("BSL", bsl.Name).Infof("Invalidating Backup Repository %s", list.Items[i].Name)
-		if err := r.patchBackupRepository(context.Background(), &list.Items[i], repoNotReady("re-establish on BSL change or create")); err != nil {
+		if err := r.patchBackupRepository(context.Background(), &list.Items[i], repoNotReady("re-establish on BSL change, create or delete")); err != nil {
 			r.logger.WithField("BSL", bsl.Name).WithError(err).Errorf("fail to patch BackupRepository %s", list.Items[i].Name)
+			continue
 		}
+		requests = append(requests, reconcile.Request{NamespacedName: types.NamespacedName{Namespace: list.Items[i].Namespace, Name: list.Items[i].Name}})
 	}
 
-	return []reconcile.Request{}
+	return requests
+}
+
+type bslPredicate struct {
```

Comment:
Let's try to reuse a create a generic predicate which could be shared with others.